simpledotorg/simple-server
# My facilities CSV download

**Story card:** [ch5103](https://app.shortcut.com/simpledotorg/story/5103/add-csv-rails-view-file)

## Checklist to get to done

* [x] clean up export test to verify specific cells
* [x] make sure the stats service returns only the data needed for the UI based my facilities actions
* [x] gate the action on a feature flag via a `before_action` ... search for `current_admin.feature_enabled?`
* [x] wrap the view code in a feature flag
* [x] make sure tests are passing and CI is passing
* [x] final review + approval + ship!


## Because

We need to provide a downloadable format of facility information for authorized personnel to access this data without the need for constant login access. This data may need to live in storage external to our database, such as for government inspection or facility reference purposes.

## This addresses

Externalizing required data without having to login for this data.

## Test instructions

Run server, log in with credentials, head to path /my_facilities/bp_controlled.csv , a download should take place. CSV file generation apply the data rendered by current filters

CSV format for this draft should match this [google sheet](https://docs.google.com/spreadsheets/d/1XHevt4SpkZoMkyX_PbQ2Bp-3ow13Pm9jN7SZpUfy5iY/edit#gid=530082164)

Here's a screen shot:

<img width="1581" alt="image" src="https://user-images.githubusercontent.com/69/138940600-3348fc36-6634-4f4c-bdab-39fb199e4ad5.png">
